### ISSUE TITLE
Using regplot() on a figure with multiple axes causes problems

### ISSUE THREAD

**Issue by dallascard** [mwaskom/seaborn#1568]
When using `regplot()` on a figure with multiple axes, the x and y limits on the last subplot are partially determined by data plotted in the other axes.

For example, the following code will produce a figure where the range of x values is way too large on the second subplot:

```import matplotlib.pyplot as plt
import seaborn
import numpy as np
import pandas as pd

np.random.seed(32)
y = np.random.randn(20, 1)
x0 = np.random.randn(20, 1) + 100
x1 = np.random.randn(20, 1)

df = pd.DataFrame(np.hstack([y, x0, x1]), columns=['y', 'x0', 'x1'])

fig, axes = plt.subplots(1, 2)
seaborn.regplot(x='x0', y='y', data=df, ax=axes[0])
seaborn.regplot(x='x1', y='y', data=df, ax=axes[1])

plt.show();
```

![regplot](https://user-images.githubusercontent.com/2933570/45769602-f87a9e80-bbf4-11e8-88fb-95a5ca88ea8f.png)

The reason seems to be that the `plot()` function in `_RegressionPlotter` uses `plt.plot()` to create (and then remove) a line, in order to determine what colour to use for the regplot (line 327 of regression.py).

`lines, = plt.plot(self.x.mean(), self.y.mean())`

This has the unintended side effect of updating the limits of one of the plots. Because this is done with `plt` instead of `ax`, this affects the last of the subplots created using `subplots()`, rather than the ax that has been passed to regplot. In the example above, because the first and second columns of X have very different means, the xlim of the second plot is pulled very far to the right.

I believe this can be fixed by changing `plt.plot()` on line 327 of regression.py to `ax.plot()`, and will submit a PR. The proposed change will also cause all subplots to be plotted in the same colour by default (unlike above). However, this will also be more consistent with `lmplot`.

Version info:
python: 3.6.1
seaborn: 0.9.0
matplotlib: 2.0.2

**Comment by piotrdzwiniel** [comment]
I've faced another problem related to the multiple regplots and axes. When one want to draw two regplots on a single axis `truncate=False` works only for the last drawn plot resulting for the first regplot to be limited by its data points. See the example code and the plot below (problem wasn't there in 0.8.1 seaborn version):

```
import pandas
import matplotlib.pyplot as plt
import seaborn

data = {
    'x_a': [-0.0045, -0.0457, 0.2457, -0.1302],
    'x_b': [-0.0826, -0.2096, 0.0031, 0.1761],
    'y_a': [-0.1527, -0.0693, 0.3608, -0.4337],
    'y_b': [-0.1337, -0.2430, 0.0155, 0.1096]
}

data_frame = pandas.DataFrame(data)

ax = plt.subplot(111)

seaborn.regplot(x='x_a', y='y_a', data=data_frame, truncate=False, ax=ax)
seaborn.regplot(x='x_b', y='y_b', data=data_frame, truncate=False, ax=ax)

plt.show()
```

![regplot](https://user-images.githubusercontent.com/7367330/48516391-8b396180-e863-11e8-968b-effe195746cc.png)

**Comment by mwaskom** [comment]
That looks unrelated

**Comment by mmagnuski** [comment]
@piotrdzwiniel I can confirm - this seems to have been introduced in seaborn 0.9 (your snippet works well on 0.8.1).

**Comment by mmagnuski** [comment]
@mwaskom - You might be correct with respect to axis limist not showing full range of data previously, sorry, I was not very thorough in checking. However I updated only seaborn, other libs stayed the same as they were. I agree it is not a bug, I just know @piotrdzwiniel and wanted to say hi this way. :)

**Comment by mwaskom** [comment]
Right. There was a change from 0.8.1 to 0.9.0, I'm just saying the 0.9.0 plot is the correct one.

**Comment by piotrdzwiniel** [comment]
@mwaskom Thanks for your thourough explanation. It seems right :) I'll play around with adjusting axes limits for the first plot before sending it as an argument to the "ax" param and we'll see. @mmagnuski Hi! :P

**Comment by scaomath** [comment]
Interesting. I am having the exact same issue with @dallascard . Manually set `set_xlim` and `set_ylim` in gca would do. But there has to be a bug somewhere I guess.

**Comment by mwaskom** [comment]
Fixed by #1569
